iceberg/patched-iceberg-core/pom.xml

```diff
@@ -0,0 +1,80 @@
+    <parent>
+        <groupId>org.apache.hive</groupId>
+        <artifactId>hive-iceberg</artifactId>
+        <version>4.0.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.iceberg</groupId>
+    <artifactId>iceberg-core</artifactId>
+    <version>patched-${iceberg-api.version}-${parent.version}</version>
+    <name>Patched Iceberg Core</name>
+
+    <!-- Temporary package until we need unreleased Iceberg changes for development purposes. -->
+    <!-- We should periodically clean this package during rebasing to newer Iceberg releases. -->
+    <!-- We should remove the package for good when Iceberg integration has been stabilized. -->
+
+    <!-- In your IDE you have to add manually the generated jar to the classpath of the -->
+    <!-- hive-iceberg-handler, hive-iceberg-catalog packages -->
+    <!-- The jar could be found here: -->
+    <!-- <HIVE_ROOT>/iceberg/patched-iceberg-core/target/iceberg-core-patched-0.11.0-4.0.0-SNAPSHOT.jar -->
+
+    <properties>
+        <hive.path.to.root>../..</hive.path.to.root>
+        <path.to.iceberg.root>..</path.to.iceberg.root>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.iceberg</groupId>
+            <artifactId>iceberg-core</artifactId>
+            <version>${iceberg-api.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.iceberg</groupId>
+            <artifactId>iceberg-common</artifactId>
+            <version>${iceberg-api.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.iceberg</groupId>
+            <artifactId>iceberg-api</artifactId>
+            <version>${iceberg-api.version}</version>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>unpack</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.apache.iceberg</groupId>
+                                    <artifactId>iceberg-core</artifactId>
+                                    <version>${iceberg-api.version}</version>
+                                    <type>jar</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${project.build.directory}/classes</outputDirectory>
+                                    <excludes>
```

Comment:
I think I'm missing some step here: don't we need to list the SnapshotSummary class to be replaced here?

Reply:
Those will come with the individual changes